gramps_webapi/api/resources/util.py

```diff
@@ -1,112 +1,246 @@
+    ]
 
 
-def get_parents_grampsids(db: DbReadBase, person: Person) -> Optional[GrampsId]:
-    """Get the Gramps IDs of the family's parents."""
-    handle = person.get_main_parents_family_handle()
-    try:
-        return db.get_family_from_handle(handle).gramps_id
-    except HandleError:
-        return None
+def get_events_for_references(db: DbReadBase, obj: GrampsObject) -> List[Event]:
+    """Get events for a Gramps Object."""
+    return [db.get_event_from_handle(event_ref.ref) for event_ref in obj.event_ref_list]
 
 
-def get_families_grampsids(db: DbReadBase, person: Person) -> List[GrampsId]:
-    """Get the Gramps IDs of all the person's families."""
-    handles = person.get_family_handle_list() or []
-    return [db.get_family_from_handle(handle).gramps_id for handle in handles]
+def get_media_for_references(db: DbReadBase, obj: GrampsObject) -> List[Media]:
+    """Get media for a Gramps Object."""
+    return [db.get_media_from_handle(media_ref.ref) for media_ref in obj.media_list]
 
 
-def get_event_grampsids_roles(db, obj: GrampsObject):
-    """Get the Gramps ID and role of events of a Gramps object."""
+def get_people_for_references(db: DbReadBase, person: Person) -> List[Person]:
+    """Get people associated with a Person."""
     return [
-        {
-            "gramps_id": db.get_event_from_handle(r.ref).gramps_id,
-            "role": r.get_role().string,
-        }
-        for r in obj.get_event_ref_list()
+        db.get_person_from_handle(person_ref.ref)
+        for person_ref in person.person_ref_list
     ]
 
 
-def get_citation_grampsids(db: DbReadBase, obj: GrampsObject) -> List[GrampsId]:
-    """Get the Gramps IDs of direct citations of a Gramps object."""
+def get_repositories_for_references(db: DbReadBase, source: Source) -> List[Repository]:
+    """Get repositories for a Source."""
     return [
-        db.get_citation_from_handle(handle).gramps_id
-        for handle in obj.get_citation_list()
+        db.get_repository_from_handle(repo_ref.ref) for repo_ref in source.reporef_list
     ]
 
 
-def get_note_grampsids(db: DbReadBase, obj: GrampsObject) -> List[GrampsId]:
-    """Get the Gramps IDs of direct notes of a Gramps object."""
-    return [db.get_note_from_handle(handle).gramps_id for handle in obj.get_note_list()]
+def get_source_by_handle(db: DbReadBase, handle: Handle) -> Source:
+    """Get a source and all extended attributes."""
+    obj = db.get_source_from_handle(handle)
+    obj.extended = {
+        "repositories": get_repositories_for_references(db, obj),
+        "media": get_media_for_references(db, obj),
+        "notes": [db.get_note_from_handle(handle) for handle in obj.note_list],
+        "tags": [db.get_tag_from_handle(handle) for handle in obj.tag_list],
+    }
+
+
+def get_sex_profile(person: Person):
+    """Get character substitution for enumerated sex."""
+    if person.gender == person.MALE:
+        return "M"
```

Comment:
These should be string constants in `const`

Reply:
Agree